### PR TITLE
email, password and hash

### PR DESCRIPTION
**Pull request by JozeFe**
- add email, password, Role to Employee
- change tests expected exception to DataAccessException for more clear code
- add password hash to EmployeeService

### CODE REVIEW

File: fleet-management-service/src/main/java/cz/fi/muni/pa165/service/JourneyServiceImpl.java [hunk 1]
```diff
@@ -31,17 +31,17 @@
 
         try {
             journeys = journeyDao.findByEmployee(employee);
-        } catch(RuntimeException ex) {
+        } catch (RuntimeException ex) {
```
Comment (Qerts) [inline fleet-management-service/src/main/java/cz/fi/muni/pa165/service/JourneyServiceImpl.java]: jiny typ nemuze nastat? z datove vrstvy by se do servisni mely dostat jen DataAccessExeptions

File: fleet-management-api/src/main/java/cz/fi/muni/pa165/dto/EmployeeDTO.java [hunk 2]
```diff
@@ -75,7 +109,7 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return "Employee{" +
+        return "EMPLOYEE{" +
```
Comment (JozeFe) [inline fleet-management-api/src/main/java/cz/fi/muni/pa165/dto/EmployeeDTO.java]: conventions for enums in java

File: fleet-management-persistence/src/main/java/cz/fi/muni/pa165/entity/VehicleCategory.java [hunk 3]
```diff
@@ -15,7 +15,6 @@
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
 
-    @NotNull
```
Comment (JozeFe) [inline fleet-management-persistence/src/main/java/cz/fi/muni/pa165/entity/VehicleCategory.java]: nie ale je to zadefinovane v @Column(nullable = false) hned riadok pod, @NotNull robilo to ze to skontroloval hibernate este pred vlozenim do databazy a vyhodil vynimku taku ze ju postProcessor nevedel prelozit na DataAccessException a neprechadzal test,

Comment (JozeFe) [inline fleet-management-persistence/src/main/java/cz/fi/muni/pa165/entity/VehicleCategory.java]: do databazy ale aj tak nemozes vlozit null name, vyhodi to DataAccessException je nato aj test :)

**Pull request merged**